synapse/config/server.py

```diff
@@ -350,7 +350,7 @@ def default_config(self, server_name, data_dir_path, **kwargs):
             resources:
               - names: [client]
                 compress: true
-              - names: [federation]
+              - names: [federation, openid]
```

Comment:
I wouldn't bother adding these if they get added by `federation`. You might want to add a separate (commented out) entry to the `resources` section (like we do below for the `manhole` listener) to document that `openid` is a thing. Maybe something like:

```yaml
# # If federation is disabled synapse can still expose the open ID endpoint
# # to allow integrations to authenticate users
# - names: [openid]
#   compress
```